src/js/types/User.ts

```diff
@@ -28,7 +38,18 @@ export interface IWritableUserMetadata {
   }
 }
 
-export type IUserMetadata = IWritableUserMetadata & IReadOnlyUserMetadata
+/**
+ * `user_metadata` as seen in the Auth0 datastore,
+ * before we add fields during post-login actions.
+ */
+export type IUserMetadataOriginal = IWritableUserMetadata & IReadOnlyUserMetadata
+
+/* `roles` are added by a post-login action and are not in
+ * the user's `user_metadata` object in Auth0.
+ */
+export type IUserMetadata = IUserMetadataOriginal & {
+  roles: UserRole[]
```

Comment:
Pull this out because this field cannot be set on the user since it's not actually in the user's `user_metadata` field.